js/functions.js

```diff
@@ -0,0 +1,89 @@
+//Задание №1
+
+function checkStringLength (string, lengthLimit) {
+  return string.length <= lengthLimit;
+}
+
+checkStringLength('Электродвигатель', 16);
+
+//Задание №2
+
+function checkPalindrome (string) {
+  let newString = '';
+
+  string = string.toLowerCase().replaceAll(' ', '');
+
+  for (let i = string.length - 1; i >= 0; i--) {
+    newString += string[i];
+  }
+
+  return newString === string;
+}
+
+checkPalindrome('топот'); //true
+checkPalindrome('ДовОд'); //true
+checkPalindrome('Кекс'); //false
+checkPalindrome('Лёша на полке клопа нашёл '); //true
+
+//Задание №3
+
+function getNumber (value) {
+  let newNumber = '';
+
+  if (typeof(value) === 'number') {
+    value = String(value);
+
+    for (let i = 0; i < value.length; i++) {
+      if (value[i] >= 0) {
+        newNumber += value[i];
+      }
+    }
+  } else {
+    value = value.replaceAll(' ', '');
+
+    for (let i = 0; i < value.length; i++) {
+      if (value[i] >= 0) {
+        newNumber += value[i];
+      }
+    }
+
+    if (newNumber === '') {
+      return NaN;
+    }
+  }
+
+  return Number(newNumber);
+}
+
+getNumber('2023 год'); //Результат: число 2023
+getNumber('ECMAScript 2022'); //Результат: число 2022
+getNumber('1 кефир, 0.5 батона'); //Результат: число 105
+getNumber('а я томат'); //Результат: NaN
+getNumber(2023); //Результат: число 2023
+getNumber(-1); //Результат: число 1
+getNumber(1.5); //Результат: число 15
+
+//Задание №4
+
+function createsFileAddresses (prymaryString, minLength, secondaryString) {
+
+  let newString = '';
+  const numberCicle = minLength - (prymaryString.length + secondaryString.length);
+
+  secondaryString = secondaryString.slice(0, minLength - prymaryString.length);
+
+  if (prymaryString.length + secondaryString.length < minLength) {
+    for (let i = 0; i < numberCicle; i++) {
+      newString += secondaryString[0];
```

Comment:
ну тут явно не 0 должен быть )
```suggestion
  if (prymaryString.length + secondaryString.length < minLength) {
    for (let i = 0; i < numberCicle; i++) {
      newString += secondaryString[i];
```

Reply:
а вообще помимо этого есть ещё проблемка, потести с наборами данных где secondaryString влезает больше одного раза, и например не целое число раз, например createsFileAddresses('q', 5, 'we'); и createsFileAddresses('q', 6, 'we');

Reply:
там что бы в цикле undefined не вылетоло, numberCicle должен ещё и длину secondaryString учитывать

Reply:
> 
Не совсем понимаю почему ноль нельзя там оставить? Там ведь идея, чтобы собирать первую букву, если основной и второстепенной строки не хватает для минимальной длины строки. И тогда такие примеры, как:
createsFileAddresses('q', 5, 'we');
createsFileAddresses('q', 6, 'we');
Полностью отработают как надо.

Reply:
Либо тогда добавить еще одну переменную в цикле j, где j = 0 и она не будет увеличиваться, просто с каждым циклом добавлять первую букву 
newString += secondaryString[j];